src/Tools/IdeBenchmarks/InheritanceMargin/InheritanceMarginServiceBenchmarks.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using BenchmarkDotNet.Attributes;
+using Microsoft.Build.Locator;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Editor.UnitTests;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.MSBuild;
+using Microsoft.CodeAnalysis.Test.Utilities;
+
+namespace IdeBenchmarks.InheritanceMargin
+{
+    [MemoryDiagnoser]
+    public class InheritanceMarginServiceBenchmarks
+    {
+        private readonly UseExportProviderAttribute _useExportProviderAttribute = new();
+        private Solution _solution;
+
+        public InheritanceMarginServiceBenchmarks()
+        {
+            _solution = null!;
+        }
+
+        [GlobalSetup]
+        public void Setup()
+        {
+            // QueryVisualStudioInstances returns Visual Studio installations on .NET Framework, and .NET Core SDK
+            // installations on .NET Core. We use the one with the most recent version.
+            var msBuildInstance = MSBuildLocator.QueryVisualStudioInstances().OrderByDescending(x => x.Version).First();
+            MSBuildLocator.RegisterInstance(msBuildInstance);
+        }
+
+        [IterationSetup]
+        public void IterationSetup()
+        {
+            _useExportProviderAttribute.Before(null);
+
+            var roslynRoot = Environment.GetEnvironmentVariable(Program.RoslynRootPathEnvVariableName);
+            var solutionPath = Path.Combine(roslynRoot, @"src\Tools\IdeCoreBenchmarks\Assets\Microsoft.CodeAnalysis.sln");
+
+            if (!File.Exists(solutionPath))
+                throw new ArgumentException("Couldn't find solution.");
+
+            Console.WriteLine("Found solution.");
+            var assemblies = MSBuildMefHostServices.DefaultAssemblies
+                .AddRange(EditorTestCompositions.EditorFeatures.Assemblies)
+                .Distinct();
+
+            var hostService = MefHostServices.Create(assemblies);
+            var workspace = MSBuildWorkspace.Create(hostService);
+            _solution = workspace.OpenSolutionAsync(solutionPath).Result;
+        }
+
+        [IterationCleanup]
+        public void IterationCleanup()
+        {
+            _useExportProviderAttribute.After(null);
+        }
+
+        [Benchmark]
+        public async Task BenchmarkInheritanceMarginServiceAsync()
```

Comment:
For services running in OOP, consider move the benchmark to IdeCoreBenchmarks. Since we support both desktop and .Net Core hosts.

Reply:
The service is now lives in EditorFeature projects and I saw these comments https://github.com/dotnet/roslyn/blob/867da8bb361131ca777e7d1af695b339a4f00e3c/src/Tools/IdeCoreBenchmarks/IdeCoreBenchmarks.csproj#L68

Not sure if I can referencing editor features directly/or the service needs to be move to Feature layer?

Reply:
The difficult part is it requires using Language Service, and so it needs this attritbute https://github.com/dotnet/roslyn/blob/main/src/Workspaces/CoreTestUtilities/MEF/UseExportProviderAttribute.cs
But it is currently targeting net6.0-windows only.

If this test is moved to IdeCoreBenchmarks, it needs reference this attribute, which cause a net6.0 project -> net6.0-windows project.